frontend/src/app/interceptor/url.interceptor.ts

```diff
@@ -6,22 +6,21 @@ import {
   HttpInterceptor, HttpHeaders
 } from '@angular/common/http';
 import { Observable } from 'rxjs';
-import { CourseServiceService } from "../services/course/course-service.service";
+import { CourseService } from "../services/course/course.service";
 
 @Injectable()
 export class URLInterceptor implements HttpInterceptor {
-  public URL = 'http://localhost:8000';
+  static URL: string = 'http://localhost:8000/api';
 
-  constructor(private readonly service: CourseServiceService) {
-  }
+  constructor(private readonly service: CourseService) { }
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     let header: HttpHeaders = request.headers.set('username', this.service.user?.name ?? 'unset')
     header = header.set('password', this.service.user?.password ?? 'unset')
     console.log(header)
     const modifiedRequest = request.clone({
       headers: header,
-      url: this.URL + request.url.startsWith('/')?'':'/' + request.url
+      url: URLInterceptor.URL + request.url.startsWith('/') ? '' : '/' + request.url
```

Comment:
warum kann hier nicht this anstelle von URLInterceptor bleiben=

Reply:
Als ich es zu `static `geändert habe und das `this` da stand kann diese error message.

`TS2576: Property  URL  does not exist on type  URLInterceptor . Did you mean to access the static member  URLInterceptor.URL  instead?`